magpie-core/src/main/java/io/openraven/magpie/core/cspm/services/ReportServiceImpl.java

```diff
@@ -27,37 +26,61 @@ private String humanReadableFormat(Duration duration) {
 
+    final Function<String, String> trimRuleName = (ruleName) -> {
+      String trimmedName = ruleName.replace(System.lineSeparator(), "");
+      trimmedName = trimmedName.length() >= COLUMN_WIDTH ? trimmedName.substring(0, COLUMN_WIDTH - "...".length() - 1) + "..." : trimmedName;
+      return trimmedName;
+    };
+
+
     System.out.println(BOLD_SET + "Scan Summary:" + BOLD_RESET);
     System.out.printf("%-30s%-40s\n", "Scan start time", this.scanMetadata.getStartDateTime().toString());
     System.out.printf("%-30s%-40s\n", "Scan duration", humanReadableFormat(this.scanMetadata.getDuration()));
-    System.out.printf("%-30s%-40d\n\n", "Total violations found", violations.size());
+    System.out.printf("%-30s%-40d\n\n", "Total violations found", results.getNumOfViolations());
+
+    if (!results.getIgnoredPolicies().isEmpty()) {
+      System.out.println(BOLD_SET + "Disabled policies:" + BOLD_RESET);
+      System.out.printf(BOLD_SET + "%-2s%-" + GUID_COLUMN_WIDTH + "s%-" + COLUMN_WIDTH + "s\n" + BOLD_RESET, "", "Policy GUID", "Policy name");
+      results.getIgnoredPolicies().forEach((policy, reason) -> {
+        System.out.printf("%-2s%-" + GUID_COLUMN_WIDTH + "s%-" + COLUMN_WIDTH + "s\n", "", policy.getPolicy().getId(), policy.getPolicy().getName());
+      });
+      System.out.printf("\n");
+    }
+
+    if (!results.getViolations().isEmpty()) {
+      System.out.println(BOLD_SET + "Scan Per-policy Details:" + BOLD_RESET);
+
+      results.getViolations().forEach((policy, policyViolations) -> {
+        System.out.printf("%-30s%-40s\n", "Policy name", policy.getPolicy().getName());
+        System.out.printf("%-30s%-40s\n", "Policy status", policy.getPolicy().isEnabled() ? "Enabled" : "Disabled");
+        System.out.printf("%-30s%-40s\n", "No. of violations", policyViolations.size());
 
-    System.out.println(BOLD_SET + "Scan Per-policy Details:" + BOLD_RESET);
-    policies.forEach(policy -> {
-      System.out.printf("%-30s%-40s\n", "Policy name", policy.getPolicy().getName());
-      System.out.printf("%-30s%-40s\n", "Policy status", policy.getPolicy().isEnabled() ? "Enabled" : "Disabled");
-      var policyViolations = violations.stream().filter(violation -> violation.getPolicyId().equals(policy.getPolicy().getId())).collect(Collectors.toList());
-      System.out.printf("%-30s%-40s\n", "No. of violations", policyViolations.size());
+        var ignoredRules = results.getIgnoredRules().get(policy);
+        if (!ignoredRules.isEmpty()) {
+          System.out.printf("%-30s\n", "Ignored rules");
+          System.out.printf(BOLD_SET + "%-2s%-" + COLUMN_WIDTH + "s%-" + GUID_COLUMN_WIDTH + "s%-" + COLUMN_WIDTH + "s\n" + BOLD_RESET, "", "Rule name", "Rule GUID", "Reason");
+          ignoredRules.forEach((rule, reason) -> {
+            System.out.printf("%-2s%-" + COLUMN_WIDTH + "s%-" + GUID_COLUMN_WIDTH + "s%-" + COLUMN_WIDTH + "s\n", "", trimRuleName.apply(rule.getName()), rule.getId(), reason);
```

Comment:
```suggestion
            System.out.printf("%-2s%-" + COLUMN_WIDTH + "s%-" + GUID_COLUMN_WIDTH + "s%-" + COLUMN_WIDTH + "s\n", "", trimRuleName.apply(rule.getName()), rule.getId(), reason.getReason());
```

Reply:
This should give the plain text reason rather than the enum constant.

Reply:
Pushed together with few more changes. Initial changes weren't perfect. e.g. ignored rules weren't printed in case all rules from policy were ignored